crates/aranya-daemon/src/config.rs

```diff
@@ -28,6 +28,13 @@ pub struct Config {
     /// Network address of Aranya sync server.
     pub sync_addr: Addr,
 
+    /// The service name used by Keyring's [`keyring::Entry`] API
+    /// See the documentation for each Platform's credential store
+    /// to see what the service name maps to.
+    ///
+    /// This value cannot be blank.
+    pub service_name: NonEmptyString,
```

Comment:
We should make the QUIC syncer config optional and extensible in case other syncer types are added in the future.
```suggestion
    /// QUIC syncer configuration
    pub quic_sync: Option<QuicSyncConfig>,
```